packages/fether-react/src/BackupPhrase/BackupPhrase.js

```diff
@@ -0,0 +1,394 @@
+      -> In this case, we do not show warning, but we alow them to decrypt the phrase with their password.
+*/
+@inject('parityStore')
+@withAccount
+class BackupPhrase extends Component {
+  state = {
+    error: null,
+    password: '',
+    phrase: '',
+    phraseRewrite: '',
+    step: 1,
+    unlocked: false
+  };
+
+  componentDidMount () {
+    this.getPhrase();
+  }
+
+  handleChangePassword = ({ target: { value } }) => {
+    this.setState({ password: value });
+  };
+
+  // handle the user typed phrase rewrite
+  handleChangePhrase = ({ target: { value } }) => {
+    this.setState({ phraseRewrite: value });
+  };
+
+  handleSubmit = async event => {
+    const {
+      account: { address },
+      history
+    } = this.props;
+    const { password, phrase } = this.state;
+
+    // unset account flag
+    await localForage.removeItem(`__flagged_${address}`);
+
+    // AES encrypt the phrase
+    const encryptedPhrase = CryptoJS.AES.encrypt(phrase, password).toString();
+
+    // set flag as safe
+    await localForage.setItem(`__safe_${address}`, encryptedPhrase);
+
+    // go to account
+    history.push(`/tokens/${address}`);
+  };
+
+  // Get the actual phrase to rewrite
+  getPhrase = async () => {
+    const {
+      account: { address }
+    } = this.props;
+
+    const phrase =
+      (await localForage.getItem(`__flagged_${address}`)) ||
+      (await localForage.getItem(`__safe_${address}`));
+
+    // phrase is still encrypted
+    this.setState({ phrase });
+  };
+
+  onCopyOrPastePhrase = event => {
+    // Disable copying the phrase if in production.
+    // Keep it enabled for development.
+    if (process.env.NODE_ENV === 'production') {
+      event.preventDefault();
+      this.setState({
+        error:
+          'Copy and pasting is disabled for this step. Please type out your full recovery phrase.'
+      });
+
+      return false;
+    }
+  };
+
+  nextStep = event => {
+    const {
+      account: { address },
+      history,
+      location: { pathname }
+    } = this.props;
+
+    event.preventDefault();
+
+    const needsRewrite = pathname.split('/')[3] === 'true';
+
+    if (!needsRewrite) {
+      history.push(`/tokens/${address}`);
+    }
+
+    this.setState({
+      step: this.state.step + 1
+    });
+  };
+
+  prevStep = () => {
+    const {
+      account: { address },
+      history
+    } = this.props;
+    const { step } = this.state;
+
+    if (step === 1) {
+      history.push(`/tokens/${address}`);
+    }
+
+    this.setState({
+      step: step - 1
+    });
+  };
+
+  // Decrypt the encrypted phrase from state
+  unlockWithPassword = async event => {
+    const { password, phrase } = this.state;
+
+    try {
+      // decrypt the phrase
+      var phraseInBytes = CryptoJS.AES.decrypt(phrase, password);
+      // this will throw in the case of a wrong password
+      var originalPhrase = phraseInBytes.toString(CryptoJS.enc.Utf8);
```

Comment:
just a thought: in case of an invalid password, the decryption might output a valid utf8 string (maybe?). depending on how likely this is, we might want to check in addition if the passphrase is a valid bip39 phrase.